iroh-net/src/magicsock.rs

```diff
@@ -742,13 +782,30 @@ impl AsyncUdpSocket for MagicSock {
                             return Poll::Ready(Err(err));
                         }
                         NetworkReadResult::Ok {
-                            bytes,
-                            meta,
+                            mut bytes,
+                            mut meta,
                             source,
                         } => {
-                            buf_out[..bytes.len()].copy_from_slice(&bytes);
+                            let copy_len = if bytes.len() > buf_out.len() {
+                                // split bytes on segment boundary, leaving the largest
```

Comment:
the splitting is already done on the IP recv path, so I would only run this path if the source is `derp`